django-s3-file-field-client/src/client.ts

```diff
@@ -114,6 +122,7 @@ export default class S3FFClient {
    * @param multipartInfo Signed information returned from /upload-complete/.
    */
   protected async finalize(multipartInfo: MultipartInfo): Promise<string> {
+    this.onProgress({ state: 'finalizing' });
     const response = await axios.post(`${this.baseUrl}/finalize/`, {
       upload_signature: multipartInfo.upload_signature,
     });
```

Comment:
Once this completes, it'd be nice to set the progress to some terminal state. I think this can also be set at the end of `uploadFile`.

Reply:
Maybe so, although at that point it's functionally equivalent to just awaiting the Promise resolution, so it doesn't add practical value. If you feel strongly though, I'll add it.

Reply:
I think it can make things simpler for implementors, depending on how they structure things. I think it's worth adding.

Reply:
Less assembly required is usually better for downstream usability.